react-components/src/components/NodeCacheProvider/Image360AnnotationCache.ts

```diff
@@ -69,10 +70,30 @@ export class Image360AnnotationCache {
           if (asset === undefined) {
             return acc;
           }
+          const centerPosition: Vector3[] = [];
+          const tranform = annotationInfo.imageEntity.transform;
+          const revisionAnnotationsPromise = annotationInfo.imageRevision.getAnnotations();
+          void revisionAnnotationsPromise.then((annotations) => {
+            annotations.forEach((revisionAnnotation) => {
+              if (
+                assetId ===
+                getAssetIdOrExternalIdFromImage360Annotation(revisionAnnotation.annotation)
+              ) {
+                const center = revisionAnnotation.getCenter().applyMatrix4(tranform);
+                if (center !== undefined) {
+                  centerPosition.push(center);
+                }
+              }
+            });
+          });
```

Comment:
This is not safe - we should not just assume that this async call will be done executing before the rest of this function is run